state-surgery/state/state_test.go

```diff
@@ -174,3 +174,232 @@ func testContractStateValuesAreEmpty(t *testing.T, contract *testdata.Testdata)
+				{
+					Key:   common.Hash{},
+					Value: common.Hash{},
+				},
+			},
+			expect: []*state.EncodedStorage{
+				{
+					Key:   common.Hash{},
+					Value: common.Hash{},
+				},
+			},
+		},
+		{
+			// Two duplicate inputs should be merged
+			input: []*state.EncodedStorage{
+				{
+					Key:   common.Hash{1},
+					Value: common.Hash{},
+				},
+				{
+					Key:   common.Hash{1},
+					Value: common.Hash{},
+				},
+			},
+			expect: []*state.EncodedStorage{
+				{
+					Key:   common.Hash{1},
+					Value: common.Hash{},
+				},
+			},
+		},
+		{
+			// Two different inputs should be the same result
+			input: []*state.EncodedStorage{
+				{
+					Key:   common.Hash{1},
+					Value: common.Hash{},
+				},
+				{
+					Key:   common.Hash{2},
+					Value: common.Hash{},
+				},
+			},
+			expect: []*state.EncodedStorage{
+				{
+					Key:   common.Hash{1},
+					Value: common.Hash{},
+				},
+				{
+					Key:   common.Hash{2},
+					Value: common.Hash{},
+				},
+			},
+		},
+		{
+			// Two matching keys should be merged bitwise
+			input: []*state.EncodedStorage{
+				{
+					Key:   common.Hash{},
+					Value: common.Hash{0x00, 0x01},
+				},
+				{
+					Key:   common.Hash{},
+					Value: common.Hash{0x02, 0x00},
+				},
+			},
+			expect: []*state.EncodedStorage{
+				{
+					Key:   common.Hash{},
+					Value: common.Hash{0x02, 0x01},
+				},
+			},
+		},
+	}
+
+	for _, test := range cases {
+		got := state.MergeStorage(test.input)
+		require.Equal(t, got, test.expect)
+	}
+}
+
+func TestEncodeUintValue(t *testing.T) {
+	cases := []struct {
+		number any
+		offset uint
+		expect common.Hash
+	}{
+		{
+			number: 0,
+			offset: 0,
+			expect: common.Hash{},
+		},
+		{
+			number: big.NewInt(1),
+			offset: 0,
+			expect: common.Hash{31: 0x01},
+		},
+		{
+			number: uint64(2),
+			offset: 0,
+			expect: common.Hash{31: 0x02},
+		},
+		{
+			number: uint8(3),
+			offset: 0,
+			expect: common.Hash{31: 0x03},
+		},
+		{
+			number: uint16(4),
+			offset: 0,
+			expect: common.Hash{31: 0x04},
+		},
+		{
+			number: uint32(5),
+			offset: 0,
+			expect: common.Hash{31: 0x05},
+		},
+		{
+			number: int(6),
+			offset: 0,
+			expect: common.Hash{31: 0x06},
+		},
+		{
+			number: 1,
+			offset: 1,
+			expect: common.Hash{30: 0x01},
+		},
+		{
+			number: 1,
+			offset: 10,
+			expect: common.Hash{21: 0x01},
+		},
+	}
+
+	for _, test := range cases {
+		got, err := state.EncodeUintValue(test.number, test.offset)
+		require.Nil(t, err)
+		require.Equal(t, got, test.expect)
+	}
+}
+
+func TestEncodeBoolValue(t *testing.T) {
+	cases := []struct {
+		boolean any
+		offset  uint
+		expect  common.Hash
+	}{
+		{
+			boolean: true,
+			offset:  0,
+			expect:  common.Hash{31: 0x01},
+		},
+		{
+			boolean: false,
+			offset:  0,
+			expect:  common.Hash{},
+		},
+		{
+			boolean: true,
+			offset:  1,
+			expect:  common.Hash{30: 0x01},
+		},
+		{
+			boolean: false,
+			offset:  1,
+			expect:  common.Hash{},
+		},
+		{
+			boolean: "true",
```

Comment:
duplicate to above

Reply:
The difference is the type of the argument, it handles strings and bools